cobigen/cobigen-core-api/src/main/java/com/devonfw/cobigen/api/util/to/Nexus2SearchResponse.java

```diff
@@ -0,0 +1,175 @@
+package com.devonfw.cobigen.api.util.to;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.devonfw.cobigen.api.constants.MavenSearchRepositoryConstants;
+import com.devonfw.cobigen.api.constants.MavenSearchRepositoryType;
+import com.devonfw.cobigen.api.exception.RESTSearchResponseException;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * Json model for nexus2 Search REST API response
+ *
+ */
+@JsonIgnoreProperties(value = { "totalCount", "from", "count", "tooManyResults", "collapsed", "repoDetails" })
+public class Nexus2SearchResponse implements SearchResponse {
+
+  /** Logger instance. */
+  @JsonIgnore
+  private static final Logger LOG = LoggerFactory.getLogger(Nexus2SearchResponse.class);
+
+  @JsonProperty("data")
+  private List<Nexus2SearchResponseData> data;
+
+  @Override
+  @JsonIgnore
+  public List<URL> getDownloadURLs() throws MalformedURLException {
+
+    List<URL> downloadLinks = new ArrayList<>();
+
+    for (Nexus2SearchResponseData item : this.data) {
+      for (Nexus2SearchResponseArtifactHits artifactHit : item.artifactHits) {
+        for (Nexus2SearchResponeArtifactLinks artifactLink : artifactHit.artifactLinks) {
+          downloadLinks.add(SearchResponseUtil.createDownloadLink(
+              MavenSearchRepositoryConstants.NEXUS2_REPOSITORY_URL + "/"
+                  + MavenSearchRepositoryConstants.NEXUS2_REPOSITORY_LINK,
+              item.getGroupId(), item.getArtifactId(), item.getVersion(), "." + artifactLink.getExtension()));
+
+        }
+      }
+    }
+
+    // removes duplicates
+    List<URL> newDownloadList = downloadLinks.stream().distinct().collect(Collectors.toList());
+
+    return newDownloadList;
+  }
+
+  @Override
+  @JsonIgnore
+  public String getJsonResponse(String repositoryUrl, String groupId) throws RESTSearchResponseException {
+
+    return getJsonResponse(repositoryUrl, groupId, null);
+  }
+
+  @Override
+  @JsonIgnore
+  public String getJsonResponse(String repositoryUrl, String groupId, String authToken)
+      throws RESTSearchResponseException {
+
+    String targetLink = repositoryUrl + "/" + MavenSearchRepositoryConstants.NEXUS2_TARGET_LINK + "?q=" + groupId;
+    LOG.info("Starting {} search REST API request with URL: {}.", getRepositoryType(), targetLink);
+
+    String jsonResponse;
+
+    jsonResponse = SearchResponseUtil.getJsonResponseStringByTargetLink(targetLink, authToken);
```

Comment:
`get` ?

Reply:
Renamed to retrieve... and moved to AbstractSearchResponse.